homeassistant/components/youtube/config_flow.py

```diff
@@ -59,54 +47,38 @@ def extra_authorize_data(self) -> dict[str, Any]:
             "prompt": "consent",
         }
 
-    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
-        """Perform reauth upon an API authentication error."""
-        self.reauth_entry = self.hass.config_entries.async_get_entry(
-            self.context["entry_id"]
-        )
-        return await self.async_step_reauth_confirm()
-
-    async def async_step_reauth_confirm(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
-        """Confirm reauth dialog."""
-        if user_input is None:
-            return self.async_show_form(step_id="reauth_confirm")
-        return await self.async_step_user()
-
     async def async_oauth_create_entry(self, data: dict[str, Any]) -> FlowResult:
         """Create an entry for the flow, or update existing entry."""
-
-        service = build(
-            "youtube",
-            "v3",
-            credentials=Credentials(data[CONF_TOKEN][CONF_ACCESS_TOKEN]),
-        )
-        # pylint: disable=no-member
-        own_channel_request: HttpRequest = service.channels().list(
-            part="snippet", mine=True
-        )
-        response = await self.hass.async_add_executor_job(own_channel_request.execute)
-        user_id = response["items"][0]["id"]
+        try:
+            service = build(
```

Comment:
My impression is that this does discovery which can send an HTTP request. This blocking call also needs to be in the executor.
I believe this is also why its structured this way in gmail:
https://github.com/home-assistant/core/blob/fc7a421a48dc6bb041a450398c70fc434b5299f9/homeassistant/components/google_mail/config_flow.py#LL61C13-L61C25

Reply:
This is also the case for `get_resource` in the api module, which i think may be a mistake in gmail that its not run in the executor.

Reply:
I'm not sure how good the blocking detection works. I have yet to see any issues with Google Mail.

Reply:
The methods do http connections. They may not in all scenarios but the api docs also mentioned raising http related exceptions and you can see when browsing the code where it makes connections in discovery.

I'm not sure how that observations of how this performs in practice come in to play. It's intended as a synchronous API.